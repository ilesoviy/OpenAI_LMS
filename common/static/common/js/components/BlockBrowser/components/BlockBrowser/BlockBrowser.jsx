/* global gettext */
import { Button, Icon } from '@edx/paragon';
import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';

const RightIcon = (
    <Icon
        className={['fa', 'fa-arrow-right']}
        screenReaderText={gettext('View child items')}
    />
);

const UpIcon = (
    <Icon
        className={['fa', 'fa-arrow-up']}
        screenReaderText={gettext('Navigate up')}
    />
);

const BLOCK_TYPE_NAME = {
    course: 'Course',
    chapter: 'Section',
    sequential: 'Sub-section',
    vertical: 'Unit',
};

const BlockType = PropTypes.shape({
    children: PropTypes.array,
    display_name: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    parent: PropTypes.string,
    type: PropTypes.string.isRequired,
});

export const BlockList = ({
    blocks, selectedBlock, onSelectBlock, onChangeRoot,
}) => (
    <ul className="block-list">
        {blocks.map(block => (
            <li
                key={block.id}
                className={classNames(`block-type-${block.type}`, { selected: block.id === selectedBlock })}
            >
                <Button
                    className={['block-name']}
                    onClick={() => onSelectBlock(block.id)}
                    label={block.display_name}
                />
                {block.children
        && (
            <Button
                onClick={() => onChangeRoot(block.id)}
                label={RightIcon}
            />
        )}
            </li>
        ))}
    </ul>
);

BlockList.propTypes = {
    blocks: PropTypes.arrayOf(BlockType),
    selectedBlock: PropTypes.string,
    onSelectBlock: PropTypes.func.isRequired,
    onChangeRoot: PropTypes.func.isRequired,
};

BlockList.defaultProps = {
    blocks: null,
    selectedBlock: null,
};

export const BlockBrowser = ({
    blocks, selectedBlock, onSelectBlock, onChangeRoot, className,
}) => !!blocks && (
    <div className={classNames('block-browser', className)}>
        <div className="header">
            <Button
                disabled={!blocks.parent}
                onClick={() => blocks.parent && onChangeRoot(blocks.parent)}
                label={UpIcon}
            />
            <span className="title">
                {gettext('Browsing')} {gettext(BLOCK_TYPE_NAME[blocks.type])} &quot;
                <a
                    href="#_"
                    onClick={(event) => {
                        event.preventDefault();
                        onSelectBlock(blocks.id);
                    }}
                    title={`${gettext('Select')} ${gettext(BLOCK_TYPE_NAME[blocks.type])}`}
                >
                    {blocks.display_name}
                </a>&quot;:
            </span>
        </div>
        <BlockList
            blocks={blocks.children}
            selectedBlock={selectedBlock}
            onSelectBlock={onSelectBlock}
            onChangeRoot={onChangeRoot}
        />
    </div>
);

BlockBrowser.propTypes = {
    blocks: BlockType,
    selectedBlock: PropTypes.string,
    onSelectBlock: PropTypes.func.isRequired,
    onChangeRoot: PropTypes.func.isRequired,
};

BlockBrowser.defaultProps = {
    blocks: null,
    selectedBlock: null,
};
