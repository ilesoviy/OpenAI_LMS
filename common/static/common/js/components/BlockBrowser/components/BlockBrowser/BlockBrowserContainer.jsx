import { connect } from 'react-redux';
import { changeRoot } from '../../data/actions/courseBlocks';
import { getActiveBlockTree } from '../../data/selectors/index';
import { BlockBrowser } from './BlockBrowser';

const mapStateToProps = state => ({
    blocks: getActiveBlockTree(state),
    selectedBlock: state.selectedBlock,
});

const mapDispatchToProps = dispatch => ({
    onChangeRoot: blockId => dispatch(changeRoot(blockId)),
});

const BlockBrowserContainer = connect(
    mapStateToProps,
    mapDispatchToProps,
)(BlockBrowser);

export default BlockBrowserContainer;
