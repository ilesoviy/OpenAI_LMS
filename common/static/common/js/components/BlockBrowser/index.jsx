import React from 'react';
import { Provider } from 'react-redux';
import BlockBrowserContainer from './components/BlockBrowser/BlockBrowserContainer';
import store from './data/store';

export const BlockBrowser = props => (
    <Provider store={store}>
        <BlockBrowserContainer {...props} />
    </Provider>
);

export default BlockBrowser;
