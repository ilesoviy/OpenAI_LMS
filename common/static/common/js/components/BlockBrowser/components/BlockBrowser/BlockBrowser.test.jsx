/* global jest,test,describe,expect */
import React from 'react';
import renderer from 'react-test-renderer';
import { BlockBrowser, BlockList } from './BlockBrowser';
import testBlockTree from './test-block-tree.json';

describe('BlockList component', () => {
    test('render with basic parameters', () => {
        const component = renderer.create(
            <BlockList
                blocks={testBlockTree.children}
                onSelectBlock={jest.fn()}
                selectedBlock={null}
                onChangeRoot={jest.fn()}
            />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});

describe('BlockBrowser component', () => {
    test('render with basic parameters', () => {
        const component = renderer.create(
            <BlockBrowser
                blocks={testBlockTree}
                onSelectBlock={jest.fn()}
                selectedBlock={null}
                onChangeRoot={jest.fn()}
            />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });

    test('render with custom classname', () => {
        const component = renderer.create(
            <BlockBrowser
                blocks={testBlockTree}
                className="some-class"
                onSelectBlock={jest.fn()}
                selectedBlock={null}
                onChangeRoot={jest.fn()}
            />,
        );
        const tree = component.toJSON();
        expect(tree).toMatchSnapshot();
    });
});
