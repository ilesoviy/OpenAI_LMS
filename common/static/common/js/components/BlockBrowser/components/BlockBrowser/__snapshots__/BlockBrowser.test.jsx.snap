// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BlockBrowser component render with basic parameters 1`] = `
<div
  className="block-browser"
>
  <div
    className="header"
  >
    <button
      className="btn"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <div>
        <span
          aria-hidden={true}
          className="fa fa-arrow-up"
          id="Icon2"
        />
        <span
          className="sr-only"
        >
          Navigate up
        </span>
      </div>
    </button>
    <span
      className="title"
    >
      Browsing
       
      Course
       "
      <a
        href="#_"
        onClick={[Function]}
        title="Select Course"
      >
        edX Demonstration Course
      </a>
      ":
    </span>
  </div>
  <ul
    className="block-list"
  >
    <li
      className="block-type-chapter"
    >
      <button
        className="block-name btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        Introduction
      </button>
      <button
        className="btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div>
          <span
            aria-hidden={true}
            className="fa fa-arrow-right"
            id="Icon2"
          />
          <span
            className="sr-only"
          >
            View child items
          </span>
        </div>
      </button>
    </li>
    <li
      className="block-type-chapter"
    >
      <button
        className="block-name btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        Example Week 1: Getting Started
      </button>
      <button
        className="btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div>
          <span
            aria-hidden={true}
            className="fa fa-arrow-right"
            id="Icon2"
          />
          <span
            className="sr-only"
          >
            View child items
          </span>
        </div>
      </button>
    </li>
    <li
      className="block-type-chapter"
    >
      <button
        className="block-name btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        Example Week 2: Get Interactive
      </button>
      <button
        className="btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div>
          <span
            aria-hidden={true}
            className="fa fa-arrow-right"
            id="Icon2"
          />
          <span
            className="sr-only"
          >
            View child items
          </span>
        </div>
      </button>
    </li>
    <li
      className="block-type-chapter"
    >
      <button
        className="block-name btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        Example Week 3: Be Social
      </button>
      <button
        className="btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div>
          <span
            aria-hidden={true}
            className="fa fa-arrow-right"
            id="Icon2"
          />
          <span
            className="sr-only"
          >
            View child items
          </span>
        </div>
      </button>
    </li>
    <li
      className="block-type-chapter"
    >
      <button
        className="block-name btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        About Exams and Certificates
      </button>
      <button
        className="btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div>
          <span
            aria-hidden={true}
            className="fa fa-arrow-right"
            id="Icon2"
          />
          <span
            className="sr-only"
          >
            View child items
          </span>
        </div>
      </button>
    </li>
    <li
      className="block-type-chapter"
    >
      <button
        className="block-name btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        holding section
      </button>
      <button
        className="btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div>
          <span
            aria-hidden={true}
            className="fa fa-arrow-right"
            id="Icon2"
          />
          <span
            className="sr-only"
          >
            View child items
          </span>
        </div>
      </button>
    </li>
  </ul>
</div>
`;

exports[`BlockBrowser component render with custom classname 1`] = `
<div
  className="block-browser some-class"
>
  <div
    className="header"
  >
    <button
      className="btn"
      disabled={true}
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <div>
        <span
          aria-hidden={true}
          className="fa fa-arrow-up"
          id="Icon2"
        />
        <span
          className="sr-only"
        >
          Navigate up
        </span>
      </div>
    </button>
    <span
      className="title"
    >
      Browsing
       
      Course
       "
      <a
        href="#_"
        onClick={[Function]}
        title="Select Course"
      >
        edX Demonstration Course
      </a>
      ":
    </span>
  </div>
  <ul
    className="block-list"
  >
    <li
      className="block-type-chapter"
    >
      <button
        className="block-name btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        Introduction
      </button>
      <button
        className="btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div>
          <span
            aria-hidden={true}
            className="fa fa-arrow-right"
            id="Icon2"
          />
          <span
            className="sr-only"
          >
            View child items
          </span>
        </div>
      </button>
    </li>
    <li
      className="block-type-chapter"
    >
      <button
        className="block-name btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        Example Week 1: Getting Started
      </button>
      <button
        className="btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div>
          <span
            aria-hidden={true}
            className="fa fa-arrow-right"
            id="Icon2"
          />
          <span
            className="sr-only"
          >
            View child items
          </span>
        </div>
      </button>
    </li>
    <li
      className="block-type-chapter"
    >
      <button
        className="block-name btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        Example Week 2: Get Interactive
      </button>
      <button
        className="btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div>
          <span
            aria-hidden={true}
            className="fa fa-arrow-right"
            id="Icon2"
          />
          <span
            className="sr-only"
          >
            View child items
          </span>
        </div>
      </button>
    </li>
    <li
      className="block-type-chapter"
    >
      <button
        className="block-name btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        Example Week 3: Be Social
      </button>
      <button
        className="btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div>
          <span
            aria-hidden={true}
            className="fa fa-arrow-right"
            id="Icon2"
          />
          <span
            className="sr-only"
          >
            View child items
          </span>
        </div>
      </button>
    </li>
    <li
      className="block-type-chapter"
    >
      <button
        className="block-name btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        About Exams and Certificates
      </button>
      <button
        className="btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div>
          <span
            aria-hidden={true}
            className="fa fa-arrow-right"
            id="Icon2"
          />
          <span
            className="sr-only"
          >
            View child items
          </span>
        </div>
      </button>
    </li>
    <li
      className="block-type-chapter"
    >
      <button
        className="block-name btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        holding section
      </button>
      <button
        className="btn"
        onBlur={[Function]}
        onClick={[Function]}
        onKeyDown={[Function]}
        type="button"
      >
        <div>
          <span
            aria-hidden={true}
            className="fa fa-arrow-right"
            id="Icon2"
          />
          <span
            className="sr-only"
          >
            View child items
          </span>
        </div>
      </button>
    </li>
  </ul>
</div>
`;

exports[`BlockList component render with basic parameters 1`] = `
<ul
  className="block-list"
>
  <li
    className="block-type-chapter"
  >
    <button
      className="block-name btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Introduction
    </button>
    <button
      className="btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <div>
        <span
          aria-hidden={true}
          className="fa fa-arrow-right"
          id="Icon2"
        />
        <span
          className="sr-only"
        >
          View child items
        </span>
      </div>
    </button>
  </li>
  <li
    className="block-type-chapter"
  >
    <button
      className="block-name btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Example Week 1: Getting Started
    </button>
    <button
      className="btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <div>
        <span
          aria-hidden={true}
          className="fa fa-arrow-right"
          id="Icon2"
        />
        <span
          className="sr-only"
        >
          View child items
        </span>
      </div>
    </button>
  </li>
  <li
    className="block-type-chapter"
  >
    <button
      className="block-name btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Example Week 2: Get Interactive
    </button>
    <button
      className="btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <div>
        <span
          aria-hidden={true}
          className="fa fa-arrow-right"
          id="Icon2"
        />
        <span
          className="sr-only"
        >
          View child items
        </span>
      </div>
    </button>
  </li>
  <li
    className="block-type-chapter"
  >
    <button
      className="block-name btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      Example Week 3: Be Social
    </button>
    <button
      className="btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <div>
        <span
          aria-hidden={true}
          className="fa fa-arrow-right"
          id="Icon2"
        />
        <span
          className="sr-only"
        >
          View child items
        </span>
      </div>
    </button>
  </li>
  <li
    className="block-type-chapter"
  >
    <button
      className="block-name btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      About Exams and Certificates
    </button>
    <button
      className="btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <div>
        <span
          aria-hidden={true}
          className="fa fa-arrow-right"
          id="Icon2"
        />
        <span
          className="sr-only"
        >
          View child items
        </span>
      </div>
    </button>
  </li>
  <li
    className="block-type-chapter"
  >
    <button
      className="block-name btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      holding section
    </button>
    <button
      className="btn"
      onBlur={[Function]}
      onClick={[Function]}
      onKeyDown={[Function]}
      type="button"
    >
      <div>
        <span
          aria-hidden={true}
          className="fa fa-arrow-right"
          id="Icon2"
        />
        <span
          className="sr-only"
        >
          View child items
        </span>
      </div>
    </button>
  </li>
</ul>
`;
